### PR TITLE
[1.20.2] Initial port work; port resource loader

### PR DESCRIPTION
**Pull request by TheGlitch76**

### CODE REVIEW

File: build-logic/src/main/java/qsl/internal/Versions.java [hunk 1]
```diff
@@ -23,27 +23,27 @@ public final class Versions {
 	/**
 	 * The QSL version
 	 */
-	public static final String QSL_VERSION = "6.1.0";
+	public static final String QSL_VERSION = "7.0.0-beta.1";
```
Comment (EnnuiL) [inline build-logic/src/main/java/qsl/internal/Versions.java]:
```suggestion
	public static final String QSL_VERSION = "7.0.0-alpha.1";
```

File: library/block/block_extensions/src/main/resources/quilt_block_extensions.accesswidener
Comment (EnnuiL) [inline library/block/block_extensions/src/main/resources/quilt_block_extensions.accesswidener:1]: Something to be noted: https://github.com/QuiltMC/quilt-mappings/pull/469 has broken the "all blocks are in `net/minecraft/block`" scheme, meaning that the buildscript should be fixed in order to account for that

**Review by EnnuiL — CHANGES_REQUESTED** [review]
First pass

File: library/block/block_extensions/src/main/resources/quilt_block_extensions.accesswidener
Comment (TheGlitch76) [reply library/block/block_extensions/src/main/resources/quilt_block_extensions.accesswidener:1]: Done

File: build-logic/src/main/java/qsl/internal/Versions.java [hunk 2]
```diff
@@ -23,27 +23,27 @@ public final class Versions {
...
 	 * The target Minecraft version.
 	 */
-	public static final MinecraftVersion MINECRAFT_VERSION = new MinecraftVersion("1.20.1");
+	public static final MinecraftVersion MINECRAFT_VERSION = new MinecraftVersion("23w32a", "1.20.2");
 
 	/**
 	 * The Minecraft versions this version of QSL is compatible with.
 	 */
-	public static final List<MinecraftVersion> COMPATIBLE_VERSIONS = versions("1.20");
+	public static final List<MinecraftVersion> COMPATIBLE_VERSIONS = versions("23w32a");
```
Comment (EnnuiL) [inline build-logic/src/main/java/qsl/internal/Versions.java]:
```suggestion
	public static final List<MinecraftVersion> COMPATIBLE_VERSIONS = versions();
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]
Second pass; third incoming later;
There's still a failing check; don't worry, you just have to run `./gradlew applyLicenses`

File: build-logic/src/main/java/qsl/internal/analysis/ClassAnalysisUtils.java [hunk 3]
```diff
@@ -40,18 +40,25 @@ public static Path getClassPath(FileSystem jarFs, String path) {
...
-				})
-				.sorted(Comparator.comparing(classNode -> classNode.name));
+		return readPackage0(Files.list(jarFs.getPath(targetPackage.replace('.', '/'))));
 	}
 
+	public static Stream<ClassNode> readPackageRecursive(FileSystem jarFs, String targetPackage) throws IOException {
+		return readPackage0(Files.walk(jarFs.getPath(targetPackage.replace('.', '/'))));
+	}
+
+	public static Stream<ClassNode> readPackage0(Stream<Path> in) {
```
Comment (LambdAurora) [inline build-logic/src/main/java/qsl/internal/analysis/ClassAnalysisUtils.java]:
```suggestion
	public static Stream<ClassNode> readPackage(Stream<Path> in) {
```

Come on, we can do better for naming than just suffix it with `0`...

File: library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/QuiltResourcePackProfile.java [hunk 4]
```diff
@@ -39,4 +40,18 @@ public interface QuiltResourcePackProfile {
 	default @NotNull ResourcePackActivationType getActivationType() {
 		return ResourcePackActivationType.NORMAL;
 	}
+
+	static ResourcePackProfile.ResourcePackFactory identityFactory(ResourcePack pack) {
```
Comment (LambdAurora) [inline library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/QuiltResourcePackProfile.java:44]: Why is this in API?

**Review by LambdAurora — CHANGES_REQUESTED** [review]

File: library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/QuiltResourcePackProfile.java (hunk 4 above)
Comment (TheGlitch76) [reply library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/QuiltResourcePackProfile.java:44]: I moved it to API after noticing that the testmods need it. Vanilla does exactly the same thing, but their implementation `BuiltinResourcePackProvider#method_52435(ResourcePack)` is `protected`

**Comment by TheGlitch76** [comment]
@EnnuiL - I can't apply licenses because the licenser is completely broken on Windows (https://github.com/QuiltMC/quilt-gradle-licenser/issues/9)

**Comment by TheGlitch76** [comment]
I'll see if I can just only commit the file update that GHA reports it wants.

**Review by EnnuiL — CHANGES_REQUESTED** [review]
please address the CI errors

File: library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/InMemoryResourcePack.java [hunk 5]
```diff
@@ -38,6 +38,7 @@
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import com.google.gson.JsonObject;
 import com.mojang.logging.LogUtils;
+import net.minecraft.SharedConstants;
```
Comment (EnnuiL) [inline library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/InMemoryResourcePack.java:41]: don't forget about checkstyle

**Comment by EnnuiL** [comment]
As mentioned on the Discord, I have the intention of taking over this PR in order to kickstart progress on QSL 1.20.2

File: library/core/networking/src/main/java/org/quiltmc/qsl/networking/api/ServerPlayNetworking.java [hunk 6]
```diff
@@ -254,7 +254,7 @@ public static void send(ServerPlayerEntity player, Identifier channelName, Packe
 		Objects.requireNonNull(channelName, "Channel name cannot be null");
 		Objects.requireNonNull(buf, "Packet byte buf cannot be null");
 
-		player.networkHandler.sendPacket(createS2CPacket(channelName, buf));
+		player.networkHandler.(createS2CPacket(channelName, buf));
```
Comment (OroArmor) [inline library/core/networking/src/main/java/org/quiltmc/qsl/networking/api/ServerPlayNetworking.java:257]:
```suggestion
		player.networkHandler.getConnection().send(createS2CPacket(channelName, buf));
```

**Review by OroArmor — CHANGES_REQUESTED** [review]

**Pull request closed without merging**